tests/utils.py

```diff
@@ -345,6 +349,38 @@ def _application_docker_container(
     container.remove(force=True)
 
 
+def find_application_pid(pid: int) -> int:
+    # Application might be run using "sh -c ...", we detect the case and return the "real" application pid
+    process = Process(pid)
+    if (
+        process.cmdline()[0].endswith("sh")
+        and process.cmdline()[1] == "-c"
+        and len(process.children(recursive=False)) == 1
+    ):
+        pid = process.children(recursive=False)[0].pid
+
+    return pid
+
+
+def assert_jvm_flags_equal(actual_jvm_flags: Optional[List], expected_jvm_flags: List) -> None:
+    """
+    When comparing JVM flags, we want to ignore the "value" field
+    as it might change in the CI due to ergonomic set flags
+    """
+    assert actual_jvm_flags is not None, f"{actual_jvm_flags} != {expected_jvm_flags}"
+
+    assert len(actual_jvm_flags) == len(expected_jvm_flags), f"{actual_jvm_flags} != {expected_jvm_flags}"
+
+    for actual_flag_dict, expected_flag_dict in zip(actual_jvm_flags, expected_jvm_flags):
+        actual_flag_value = actual_flag_dict.pop("value")
+        expected_flag_value = expected_flag_dict.pop("value")
+
+        if expected_flag_value is not None:
+            assert actual_flag_value == expected_flag_value, f"{actual_jvm_flags} != {expected_jvm_flags}"
```

Comment:
I suppose your assert message should show the flag values in it, not the flags themselves. u already popped the values

Reply:
```python
        if expected_flag_value is not None:
            assert (
                actual_flag_value == expected_flag_value
            ), f"{actual_flag_dict|{'value': actual_flag_value}} != {expected_flag_dict|{'value': expected_flag_value}}"
            ```
so that if the value is not the same we will know which flag it belonged to